File: app/src/main/java/org/mifos/selfserviceapp/ui/activities/RegistrationActivity.java
package org.mifos.selfserviceapp.ui.activities;

import android.os.Bundle;

import org.mifos.selfserviceapp.R;
import org.mifos.selfserviceapp.ui.activities.base.BaseActivity;
import org.mifos.selfserviceapp.ui.fragments.RegistrationFragment;

public class RegistrationActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_registration);
        replaceFragment(RegistrationFragment.newInstance(), false, R.id.container);
    }
}

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationActivity.kt
package org.mifos.mobile.ui.registration

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.WindowInsets
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.compose.runtime.DisposableEffect
import androidx.compose.ui.platform.LocalView
import androidx.core.view.WindowCompat
import androidx.navigation.compose.rememberNavController
import org.mifos.mobile.core.ui.theme.MifosMobileTheme
import org.mifos.mobile.feature.registration.navigation.RegistrationNavGraph
import org.mifos.mobile.feature.registration.navigation.RegistrationScreen
import org.mifos.mobile.ui.activities.base.BaseActivity
import org.mifos.mobile.ui.login.LoginActivity

class RegistrationActivity : BaseActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        WindowCompat.setDecorFitsSystemWindows(window, false)
        enableEdgeToEdge()
        setContent {
            MifosMobileTheme {
                val navController = rememberNavController()
                RegistrationNavGraph(
                    startDestination = RegistrationScreen.Registration.route,
                    navController = navController,
                    navigateBack = { this.finish() },
                    startLoginActivity = { startLoginActivity() }
                )
            }
        }
    }

    private fun startLoginActivity()
    {
        this.startActivity(Intent(this, LoginActivity::class.java))
        this.finish()
    }
}
